presto/cluster/slurm_functions.sh

```diff
@@ -0,0 +1,232 @@
+
+    run_coord_image "/opt/presto-server/bin/launcher run" "coord"
+}
+
+# Runs a worker on a given node with custom configuration files which are generated as necessary.
+function run_worker {
+    [ $# -ne 4 ] && echo_error "$0 expected arguments 'gpu_id', 'worker_type', 'node_id', and 'worker_id'"
+    validate_environment_preconditions LOGS CONFIGS WORKSPACE COORD SINGLE_NODE_EXECUTION CUDF_LIB DATA
+
+    local gpu_id=$1
+    local worker_type=$2
+    local node=$3
+    local worker_id=$4
+    echo "running worker ${worker_id} of type ${worker_type} on node ${node} with gpu_id ${gpu_id}"
+    [ "$worker_type" != "cpu" ] && [ "$worker_type" != "gpu" ] && echo_error "worker type must be gpu/cpu"
+    if [ "$worker_type" == "cpu" ]; then
+	NUM_DRIVERS=64
+    else
+	NUM_DRIVERS=4
+    fi
+
+    local worker_image="${WORKSPACE}/presto-native-worker-${worker_type}.sqsh"
+    [ ! -f "${worker_image}" ] && echo_error "worker image does not exist at ${worker_image}"
+
+    # Make a copy of the worker config that can be given a unique id for this worker.
+    rm -rf "${CONFIGS}/etc_worker_${worker_id}"
+    cp -r "${CONFIGS}/etc_worker" "${CONFIGS}/etc_worker_${worker_id}"
+    local worker_config="${CONFIGS}/etc_worker_${worker_id}/config_native.properties"
+    local worker_node="${CONFIGS}/etc_worker_${worker_id}/node.properties"
+    local worker_data="${WORKSPACE}/worker_data_${worker_id}"
+
+    # Create unique configuration/data files for each worker:
+    # Give each worker a unique port.
+    sed -i "s+http-server\.http\.port.*+http-server\.http\.port=900${worker_id}+g" ${worker_config}
+    # Update discovery based on which node the coordinator is running on.
+    sed -i "s+discovery\.uri.*+discovery\.uri=http://${COORD}:8080+g" ${worker_config}
+    sed -i "s+single-node-execution-enabled.*+single-node-execution-enabled=${SINGLE_NODE_EXECUTION}+g" ${worker_config}
+    sed -i "s+task.max-drivers-per-task.*+task.max-drivers-per-task=${NUM_DRIVERS}+g" ${worker_config}
+
+    # Give each worker a unique id.
+    sed -i "s+node\.id.*+node\.id=worker_${worker_id}+g" ${worker_node}
+
+    # Create unique data dir per worker.
+    mkdir -p ${worker_data}
+    mkdir -p ${WORKSPACE}/.hive_metastore
+
+    # Run the worker with the new configs.
+    CUDA_VISIBLE_DIVICES=${gpu_id} srun -N1 -w $node --ntasks=1 --overlap \
+--container-image=${worker_image} \
+--container-mounts=${WORKSPACE}:/workspace,\
+${DATA}:/data,\
+${CONFIGS}/etc_common:/opt/presto-server/etc,\
+${worker_node}:/opt/presto-server/etc/node.properties,\
+${worker_config}:/opt/presto-server/etc/config.properties,\
+${worker_data}:/var/lib/presto/data,\
+${WORKSPACE}/.hive_metastore:/var/lib/presto/data/hive/metastore \
+--container-env=LD_LIBRARY_PATH="$CUDF_LIB:$LD_LIBRARY_PATH" \
+-- bash -lc "presto_server --etc-dir=/opt/presto-server/etc" > ${LOGS}/worker_${worker_id}.log 2>&1 &
+}
+
+# Run a cli node that will connect to the coordinator and run queries that setup a
```

Comment:
Is there a reason not to reuse the benchmarking and integration testing  scripts instead of these CLI based functions?

Similar comment applies to the result set related functions.